client-v2/src/shared/actions/ArticleFragments.ts

```diff
@@ -112,23 +115,62 @@ const replaceGroupArticleFragments = (
   }
 });
 
-function addArticleFragment(id: string) {
-  return (dispatch: Dispatch) =>
-    getArticles([id])
-      .then(([article]) => {
-        dispatch(externalArticleActions.fetchSuccess([article]));
-        const fragment = createFragment(article.id);
-        dispatch(
-          articleFragmentsReceived({
-            [fragment.uuid]: fragment
-          })
-        );
-        return fragment;
+/**
+ * Given a resource id, create an article fragment and add it to the store.
+ * The resource id can be a few different things:
+ *  - a article, tag or section (either the full URL or the ID)
+ *  - an external link.
+ */
+function createArticleFragment(
+  resourceId: string
+): ThunkResult<Promise<ArticleFragment | undefined>> {
+  return async (dispatch: Dispatch) => {
+    const isURL = isValidURL(resourceId);
+    const id = isURL ? getIdFromURL(resourceId) : resourceId;
+    let fragment: ArticleFragment;
+    if (isURL && !id) {
```

Comment:
I think we may be able to remove some of the if else's here. We could do something like this: 

```
if (!id && !isURL) {
  return;
} 

if (isUrl && !id) {
...

``` 

It's not a big deal but might make the code more readable?